src/components/activity/ActivityRegistrator.tsx

```diff
@@ -0,0 +1,188 @@
+    // Check if custom text has been entered
+    const entryText = choise.isDefault ? activityText : choise.value;
+    
+    // Create entry from information entered by user
+    const entry = {
+      text: entryText,
+      icon: route.params.icon,
+      person: '', // TODO: link to value based on choise
+      meaningful: importance,
+      entertaining: enjoyment,
+    };
+
+    // If toTime returns 0 hours its the next day, count it as 24.
+    let toHour = toTime.getHours();
+    if (toHour === 0) {
+      toHour = 24;
+    }
+
+    // Splitting at 'T' to extract only the date from the ISO string.
+    const isoDateString = date.toISOString().split('T')[0];
+
+    // Add entry att every applicable hour
+    for(let i = fromTime.getHours(); i < toHour; ++i) {
+      modifyActivities.add(isoDateString, i, entry);
+    }
+
+    // Go back
+    navigation.navigate('Activities', {activityRegistered: true})
+  };
+
+
+  const onCancel = () => {
+    // Go back
+    navigation.navigate('Activities', {activityRegistered: false})
+  };
+
+  // Attempts to dissmiss the keyboard when the 
+  const dismissKeyboard = () => {
+    Keyboard.dismiss();
+  };
+  
+  return (
+    <TouchableWithoutFeedback onPress={dismissKeyboard} style={{backgroundColor: 'red'}}>
+      <View style={{ padding: 10, flexDirection: 'column', height: '100%',  justifyContent: 'space-evenly'}}>
+
+        <View style={{ flexDirection: 'row', alignItems: 'center' }}>
+          <View style={{flex: 1, flexGrow: 1}}>
+            <Avatar.Icon icon={route.params.icon} size={iconSizes.avatar} />
+          </View>
+          <DatePicker date={date} setDate={setDate}  />
+        </View>
+
+        <View style={{ flexDirection: 'row' }}>
+          <TimePicker now={new Date()} defaultTimeOffset={60} steps={steps} fromTime={fromTime} setFromTime={setFromTime} 
+            toTime={toTime} setToTime={setToTime} />
+        </View>
+
+        <KeyboardAvoidingView>
+          <SuggestiveTextInput label={lang.activityRegistratorTextInputLabel} textInputText={activityText} setTextInputText={setActivityText} 
+            choises={ choises } choise={choise} setChoise={setChoise} />
+        </KeyboardAvoidingView>
+
+
+        <View style={{flexDirection: 'row', justifyContent: 'center'}}>
+
+          <View style={{flexDirection: 'column', width: '80%'}}>
+            
+            <View style={{ flexDirection: 'row'}}>
+              <Text>{lang.activityRegistratorImporanceLabel + ": "}</Text>
+              <Text>{importance}</Text>
+            </View>
+
+            <View style={{ flexDirection: 'row', alignItems: 'center'}}>
+              <Text>0</Text>
+              <Slider
+                style={{flex: 1}} value={5} step={1}
+                minimumValue={0} maximumValue={10}
+                onValueChange={(value: number) => {setImportance(value)}}
+                minimumTrackTintColor={colors.accent} maximumTrackTintColor="#000000"
+              />
```

Comment:
I feel like the slider should be a bit bigger.

Reply:
What do you mean with bigger? Thicker, longer?

Reply:
There seems to be no easy way to modify this. I think this will either require modifications to the library code or loading a custom thumb image and using that (which is a modifiable prop).